### PR TITLE
Hiding implementation details for lap, clustering, spectral, and label

### PR DESCRIPTION
**Pull request by cjnolet**
Also managed to remove the raft host/device buffers in the process

### CODE REVIEW

**Comment by cjnolet** [comment]
I realize I have a lot of copyrights to update :-)

File: cpp/test/cluster_solvers.cu [hunk 1]
```diff
@@ -60,7 +62,12 @@ TEST(Raft, ModularitySolvers)
   using value_type = double;
 
   handle_t h;
-  ASSERT_EQ(0, h.get_device());
+  ASSERT_EQ(0,
```
Comment (divyegala) [inline cpp/test/cluster_solvers.cu:65]: Did something go wrong here with the formatting?

File: cpp/test/eigen_solvers.cu [hunk 2]
```diff
@@ -34,7 +36,12 @@ TEST(Raft, EigenSolvers)
   using value_type = double;
 
   handle_t h;
-  ASSERT_EQ(0, h.get_device());
+  ASSERT_EQ(0,
```
Comment (divyegala) [inline cpp/test/eigen_solvers.cu:39]: Same as above

File: cpp/test/eigen_solvers.cu [hunk 3]
```diff
@@ -75,7 +82,12 @@ TEST(Raft, SpectralSolvers)
...
```
Comment (divyegala) [inline cpp/test/eigen_solvers.cu:85]: Same as above

**Review by divyegala — APPROVED** [review]
Looks very nice! Just a formatting question

File: cpp/test/eigen_solvers.cu (hunk 3 above)
Comment (cjnolet) [reply cpp/test/eigen_solvers.cu:85]: Ooh, this one is strange. I'm not sure why clang-format did this.

**Comment by cjnolet** [comment]
rerun tests

**Comment by cjnolet** [comment]
@gpucibot merge

**Pull request merged**